core/templates/dev/head/base.js

```diff
@@ -95,6 +97,11 @@ oppia.controller('Base', [
 
     // If this is nonempty, the whole page goes into 'Loading...' mode.
     $rootScope.loadingMessage = '';
+    // Method to call windowSize on resizing the window
+    $window.onresize = function() {
```

Comment:
Consider the scenario of 10 functions, It will mean 10 calls versus one in rootScope method. In a development environment making a hook function is sensible while in the case of production environment I beg to differ.